babel-worker.cjs

```diff
@@ -0,0 +1,39 @@
+const { transformSync } = require("@babel/core");
+const { mkdirSync, statSync, readFileSync, writeFileSync } = require("fs");
+const { dirname } = require("path");
+const chalk = require("chalk");
+const fancyLog = require("fancy-log");
+
+function needCompile(src, dest) {
+  let destStat;
+  try {
+    destStat = statSync(dest);
+  } catch (err) {
+    if (err.code === "ENOENT") {
+      return true;
+    } else {
+      throw err;
+    }
+  }
+  const srcStat = statSync(src);
+  return srcStat.mtimeMs > destStat.mtimeMs;
```

Comment:
were you able to test watching/incremental, wondering how effective this is? wondering whether we can reuse chokidar here too or is that slow?

Reply:
We can combine it with an incremental watch pipeline, so only changed files are compiled. I can land it in a separate PR.

The stat comparison here is essentially a replacement for `gulp-newer` so when we run `gulp build-no-bundle` when we already have `lib/**`, we don't re-compile the compiled sources.